src/emulator/gui/include/gui/settings_dialog.h

```diff
@@ -0,0 +1,61 @@
+#include <util/log.h>
+
+namespace gui {
+
+static const char *LIST_MODULES[] = {
+    "activity_db", "adhoc_matching", "apputil", "apputil_ext", "audiocodec", "avcdec_for_player", "bgapputil", "bXCe",
+    "common_gui_dialog", "dbrecovery_utility", "dbutil", "friend_select", "incoming_dialog", "ini_file_processor",
+    "libatrac", "libc", "libcdlg", "libcdlg_calendar_review", "libcdlg_cameraimport", "libcdlg_checkout", "libcdlg_companion",
+    "libcdlg_compat", "libcdlg_cross_controller", "libcdlg_friendlist", "libcdlg_friendlist2", "libcdlg_game_custom_data",
+    "libcdlg_game_custom_data_impl", "libcdlg_ime", "libcdlg_invitation", "libcdlg_invitation_impl", "libcdlg_main", "libcdlg_msg",
+    "libcdlg_near", "libcdlg_netcheck", "libcdlg_npeula", "libcdlg_npprofile2", "libcdlg_np_message", "libcdlg_np_sns_fb",
+    "libcdlg_np_trophy_setup", "libcdlg_photoimport", "libcdlg_photoreview", "libcdlg_pocketstation", "libcdlg_remote_osk",
+    "libcdlg_savedata", "libcdlg_twitter", "libcdlg_tw_login", "libcdlg_videoimport", "libclipboard", "libdbg", "libfiber",
+    "libfios2", "libg729", "libgameupdate", "libhandwriting", "libhttp", "libime", "libipmi_nongame", "liblocation",
+    "liblocation_extension", "liblocation_factory", "liblocation_internal", "libmln", "libmlnapplib", "libmlndownloader",
+    "libnaac", "libnet", "libnetctl", "libngs", "libpaf", "libpaf_web_map_view", "libperf", "libpgf", "libpvf", "librudp",
+    "libsas", "libsceavplayer", "libSceBeisobmf", "libSceBemp2sys", "libSceCompanionUtil", "libSceDtcpIp", "libSceFt2",
+    "libscejpegarm", "libscejpegencarm", "libSceJson", "libscemp4", "libSceMp4Rec", "libSceMusicExport", "libSceNearDialogUtil",
+    "libSceNearUtil", "libScePhotoExport", "libScePromoterUtil", "libSceScreenShot", "libSceShutterSound", "libSceSqlite",
+    "libSceTelephonyUtil", "libSceTeleportClient", "libSceTeleportServer", "libSceVideoExport", "libSceVideoSearchEmpr",
+    "libSceXml", "libshellsvc", "libssl", "libsystemgesture", "libult", "libvoice", "libvoiceqos", "livearea_util",
+    "mail_api_for_local_libc", "near_profile", "notification_util", "np_activity", "np_activity_sdk", "np_basic",
+    "np_commerce2", "np_common", "np_common_ps4", "np_friend_privacylevel", "np_kdc", "np_manager", "np_matching2",
+    "np_message", "np_message_contacts", "np_message_dialog_impl", "np_message_padding", "np_party", "np_ranking",
+    "np_signaling", "np_sns_facebook", "np_trophy", "np_tus", "np_utility", "np_webapi", "party_member_list",
+    "psmkdc", "pspnet_adhoc", "signin_ext", "sqlite", "store_checkout_plugin", "trigger_util", "web_ui_plugin"
+};
+
+static const char *LIST_SYS_LANG[] = {
+    "Japanese", "American English", "French", "Spanish", "German", "Italian", "Dutch", "Portugal Portuguese", "Russian", "Korean",
+    "Traditional Chinese", "Simplified Chinese", "Finnish", "Swedish", "Danish", "Norwegian", "Polish", "Brazil Portuguese", "British English", "Turkish"
+};
+} // namespace gui
```

Comment:
Looks like this file is private to the settings dialog. It could get moved out of the public header folder and into `gui/src`.

Or since it's only used from `settings_dialog.cpp` just inline it.

Reply:
dev no have wanted i put it inside cpp file, and for using this, i am obligatory need use static const char, so i have thinks put it inside header for setting dialogs, and get coede inside cpp very more clear like that :)